### ISSUE TITLE
Tests not completing on v0.5

### ISSUE THREAD

**Issue by ranjanan** [JuliaStats/Distributions.jl#507]
It kinda gets stuck at the Weibull tests. 

```
From worker 5:	    testing Truncated(Weibull(5.0, 2.0),3,5)
From worker 5:	
```

**Comment by andreasnoack** [comment]
Locally?

**Comment by jmxpearson** [comment]
I think this is the same thing I mentioned on Saturday. Looks like tests are passing on v0.4, just not nightly. Lots of warnings, one of which #506 fixes.

**Comment by jmxpearson** [comment]
I'm just ran this locally with `coverage=true`. 0.4.6 takes only a few minutes to complete, just like on Travis, while current 0.5 took > 2 hrs. This running on my 8 core machine. Without coverage, both sets of tests run in ~ 1 minute.

Basically, testing `MvNormal` and `MvLogNormal` takes *forever*. I looked at the coverage report, and the differences for `mvnormal.jl` are pretty striking (see below). Any idea what could be causing this?

(line, count):

v0.4.6
```
158-element Array{Tuple{Int64,Any},1}:
 (29,0)       
 (32,0)       
 (33,0)       
 (35,12)      
 (37,16001602)
 (39,12)      
 (41,16001602)
 (44,12)      
 (45,12)      
 (46,12)      
 (47,12000000)
 (49,12)      
 (52,6)       
 (63,25)      
 (80,10)      
 (81,10)      
 (84,3)       
 (86,2)       
 (87,1)       
 (88,3)       
 (90,1)       
 (91,1)       
 (92,1)       
 ⋮            
 (319,1)      
 (321,1)      
 (322,1)      
 (323,1)      
 (324,200)    
 (325,200)    
 (326,600)    
 (328,200)    
 (330,1)      
 (334,1)      
 (335,1)      
 (336,1)      
 (338,1)      
 (339,1)      
 (340,1)      
 (342,1)      
 (343,1)      
 (344,200)    
 (345,200)    
 (346,200)    
 (347,600)    
 (349,200)    
 (351,1)
```

v0.5-dev
```
158-element Array{Tuple{Int64,Any},1}:
 (29,0)       
 (32,0)       
 (33,0)       
 (35,24)      
 (37,32003204)
 (39,24)      
 (41,32003204)
 (44,24)      
 (45,36)      
 (46,48000060)
 (47,84000000)
 (49,24)      
 (52,12)      
 (63,50)      
 (80,40)      
 (81,20)      
 (84,6)       
 (86,4)       
 (87,2)       
 (88,6)       
 (90,2)       
 (91,2)       
 (92,2)       
 ⋮            
 (319,2)      
 (321,2)      
 (322,2)      
 (323,805)    
 (324,400)    
 (325,3400)   
 (326,3600)   
 (328,800)    
 (330,2)      
 (334,6)      
 (335,2)      
 (336,4)      
 (338,2)      
 (339,2)      
 (340,2)      
 (342,2)      
 (343,805)    
 (344,800)    
 (345,400)    
 (346,3400)   
 (347,3600)   
 (349,800)    
 (351,2)
```

**Comment by StefanKarpinski** [comment]
Accurate coverage requires turning off inlining, which kills performance.

**Comment by jmxpearson** [comment]
I understand that the inline will slow things down appreciably, but why so much slower on 0.5?

**Comment by andreasnoack** [comment]
@carnaval knew why 0.5 was so slow and has promised to find a way to make it fast again.

**Comment by andreasnoack** [comment]
I don't think so. I was told that, on 0.5, when inlining is turned off all function call are resolved dynamically. This made something simpler and it was probably conjectured that the slowdown didn't matter because inlining should be on in real use but it was probably forgotten that almost all tests are run without inlining.

**Comment by nalimilan** [comment]
Also note that deprecations slow down execution a lot.

**Comment by jmxpearson** [comment]
I think all the deprecations have been resolved (or are in current PRs).
The problem is that Distributions is making 10^7 or so calls to inline
functions. It's probably an edge case in that way.

**Comment by carnaval** [comment]
https://github.com/JuliaLang/julia/pull/17185#partial-pull-merging

**Comment by andreasnoack** [comment]
Thanks for the link. I don't think I've seen the connection to this issue from the title.

**Comment by jmxpearson** [comment]
Closed by JuliaLang/julia#17185
